packages/flutter_tools/lib/src/base/net.dart

```diff
@@ -4,152 +4,175 @@
 
 import 'dart:async';
 
-import '../base/context.dart';
+import 'package:meta/meta.dart';
+import 'package:platform/platform.dart';
+
 import '../convert.dart';
-import '../globals.dart' as globals;
 import 'common.dart';
 import 'file_system.dart';
 import 'io.dart';
+import 'logger.dart';
 
 const int kNetworkProblemExitCode = 50;
 
 typedef HttpClientFactory = HttpClient Function();
 
 typedef UrlTunneller = Future<String> Function(String url);
 
-/// Download a file from the given URL.
-///
-/// If a destination file is not provided, returns the bytes.
-///
-/// If a destination file is provided, streams the bytes to that file and
-/// returns an empty list.
-///
-/// If [maxAttempts] is exceeded, returns null.
-Future<List<int>> fetchUrl(Uri url, {
-  int maxAttempts,
-  File destFile,
-}) async {
-  int attempts = 0;
-  int durationSeconds = 1;
-  while (true) {
-    attempts += 1;
-    _MemoryIOSink memorySink;
-    IOSink sink;
-    if (destFile == null) {
-      memorySink = _MemoryIOSink();
-      sink = memorySink;
-    } else {
-      sink = destFile.openWrite();
-    }
+class Net {
+  Net({
+    HttpClientFactory httpClientFactory,
+    @required Logger logger,
+    @required Platform platform,
+  }) :
```

Comment:
uber-nit: I don't love the formatting here. Of course, if you do it the other way then you end up odd indentation on the next lines.

Reply:
Yup. I was trying to avoid the odd indentation for the ':'